VSRAD.Package/Server/CommunicationChannel.cs

```diff
@@ -81,11 +107,56 @@ public CommunicationChannel(SVsServiceProvider provider, IProject project)
             _project = project;
             _project.RunWhenLoaded((options) =>
                 options.PropertyChanged += (s, e) => { if (e.PropertyName == nameof(options.ActiveProfile)) ForceDisconnect(); });
+            var match = _extensionVersionRegex.Match(typeof(IProject).Assembly.CodeBase);
+            if (!match.Success)
+                throw new Exception("Error while getting current extension version.");
+            _extensionVersion = new Version(match.Groups["version"].Value);
         }
 
         public Task<T> SendWithReplyAsync<T>(ICommand command) where T : IResponse =>
             SendWithReplyAsync<T>(command, tryReconnect: true);
 
+        private async Task<bool> TryProcessClientHandshake(TcpClient client)
```

Comment:
This method doesn't need to return `bool`, since result is not checked and it returns only `true` or exception. I would go with simple `Task` here. Also, Visual Studio gives warning, since all async methods are expected to have an `Async` suffix in name, so I would suggest renaming this method to `TryProcessClientHandshakeAsync`

Reply:
done